src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -414,12 +416,21 @@ private function readProperty(array $zval, string $property, bool $ignoreInvalid
                         throw $e;
                     }
                 } elseif (PropertyReadInfo::TYPE_PROPERTY === $type) {
-                    if (!isset($object->$name) && !\array_key_exists($name, (array) $object)) {
+                    $valueSet = false;
+                    $useBypass = $this->byPassHooksOnProperty & self::BYPASS_HOOKS_ON_PROPERTY_READ && $access->hasHook() && !$access->isVirtual();
+                    $valueSeemsToBeNotSet = !isset($object->$name) && !\array_key_exists($name, (array) $object);
+                    if ($valueSeemsToBeNotSet || $useBypass) {
                         try {
                             $r = new \ReflectionProperty($class, $name);
+                            if ($valueSeemsToBeNotSet) {
+                                if ($r->isPublic() && !$r->hasType()) {
+                                    throw new UninitializedPropertyException(\sprintf('The property "%s::$%s" is not initialized.', $class, $name));
+                                }
+                            }
 
-                            if ($r->isPublic() && !$r->hasType()) {
-                                throw new UninitializedPropertyException(\sprintf('The property "%s::$%s" is not initialized.', $class, $name));
+                            if ($useBypass) {
+                                $result[self::VALUE] = $r->getRawValue($object);
```

Comment:
We must test that `getRawValue` method exist to be backward compatible, you can use `\PHP_VERSION_ID >= 80400` I think